src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -603,6 +603,18 @@ public static bool IsAttributeNameContext(this SyntaxTree syntaxTree, int positi
             return syntaxTree.IsTypeContext(position, cancellationToken, semanticModelOpt);
         }
 
+        public static bool IsNamespaceDeclarationNameContext(this SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
+        {
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
```

Comment:
Oh, good catch! It doesn't show up in the simple case, but if we have nested namespaces...
```
namespace Microsoft
{
	namespace Code$$Analysis
	{
	}
}
```
...then it starts suggesting top level namespaces! I think this deserves a few additional tests.